File: generators/mortalInstruments.py
__all__ = ['mortalInstruments']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm3', 'nm2'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
            var.put('nl', (var.get('nm4').get(var.get('rnd2'))+var.get('nm5').get(var.get('rnd3'))))
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', ((var.get('nm2').get(var.get('rnd'))+Js(' '))+var.get('nl')))
            else:
                if PyJsStrictEq(var.get('tp'),Js(2.0)):
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                    var.put('names', ((var.get('nm3').get(var.get('rnd'))+Js(' '))+var.get('nl')))
                else:
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                    var.put('names', ((var.get('nm1').get(var.get('rnd'))+Js(' '))+var.get('nl')))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Aaron'), Js('Adam'), Js('Aidan'), Js('Aiden'), Js('Alex'), Js('Alexander'), Js('Alfie'), Js('Andrew'), Js('Anthony'), Js('Archie'), Js('Arthur'), Js('Ashton'), Js('Bailey'), Js('Ben'), Js('Benjamin'), Js('Billy'), Js('Blake'), Js('Bobby'), Js('Bradley'), Js('Brandon'), Js('Caleb'), Js('Callum'), Js('Cameron'), Js('Charles'), Js('Charlie'), Js('Christopher'), Js('Cody'), Js('Connor'), Js('Corey'), Js('Daniel'), Js('David'), Js('Declan'), Js('Dexter'), Js('Dominic'), Js('Dylan'), Js('Edward'), Js('Elliot'), Js('Ellis'), Js('Ethan'), Js('Evan'), Js('Ewan'), Js('Finlay'), Js('Finley'), Js('Frankie'), Js('Freddie'), Js('Frederick'), Js('Gabriel'), Js('George'), Js('Harley'), Js('Harrison'), Js('Harry'), Js('Harvey'), Js('Hayden'), Js('Henry'), Js('Isaac'), Js('Jack'), Js('Jackson'), Js('Jacob'), Js('Jake'), Js('James'), Js('Jamie'), Js('Jay'), Js('Jayden'), Js('Jenson'), Js('Joe'), Js('Joel'), Js('John'), Js('Jonathan'), Js('Jordan'), Js('Joseph'), Js('Josh'), Js('Joshua'), Js('Jude'), Js('Kai'), Js('Kayden'), Js('Kian'), Js('Kieran'), Js('Kyle'), Js('Leo'), Js('Leon'), Js('Lewis'), Js('Liam'), Js('Logan'), Js('Louie'), Js('Louis'), Js('Luca'), Js('Lucas'), Js('Luke'), Js('Mason'), Js('Matthew'), Js('Max'), Js('Michael'), Js('Morgan'), Js('Nathan'), Js('Nicholas'), Js('Noah'), Js('Oliver'), Js('Ollie'), Js('Oscar'), Js('Owen'), Js('Patrick'), Js('Peter'), Js('Reece'), Js('Reuben'), Js('Rhys'), Js('Riley'), Js('Robert'), Js('Rory'), Js('Ryan'), Js('Sam'), Js('Samuel'), Js('Scott'), Js('Sean'), Js('Sebastian'), Js('Spencer'), Js('Stanley'), Js('Taylor'), Js('Theo'), Js('Thomas'), Js('Toby'), Js('Tom'), Js('Tommy'), Js('Tyler'), Js('William'), Js('Zac'), Js('Zachary'), Js('Zak')]))
var.put('nm2', Js([Js('Abbie'), Js('Abby'), Js('Abigail'), Js('Aimee'), Js('Alexandra'), Js('Alice'), Js('Alicia'), Js('Alisha'), Js('Amber'), Js('Amelia'), Js('Amelie'), Js('Amy'), Js('Anna'), Js('Ava'), Js('Bella'), Js('Bethany'), Js('Brooke'), Js('Caitlin'), Js('Cerys'), Js('Charlie'), Js('Charlotte'), Js('Chelsea'), Js('Chloe'), Js('Courtney'), Js('Daisy'), Js('Danielle'), Js('Demi'), Js('Eleanor'), Js('Eliza'), Js('Elizabeth'), Js('Ella'), Js('Ellie'), Js('Eloise'), Js('Elsie'), Js('Emilia'), Js('Emily'), Js('Emma'), Js('Erin'), Js('Esme'), Js('Eva'), Js('Eve'), Js('Evelyn'), Js('Evie'), Js('Faith'), Js('Freya'), Js('Georgia'), Js('Georgina'), Js('Grace'), Js('Gracie'), Js('Hannah'), Js('Harriet'), Js('Heidi'), Js('Hollie'), Js('Holly'), Js('Imogen'), Js('Isabel'), Js('Isabella'), Js('Isabelle'), Js('Isla'), Js('Isobel'), Js('Jade'), Js('Jasmine'), Js('Jennifer'), Js('Jessica'), Js('Jodie'), Js('Julia'), Js('Kate'), Js('Katherine'), Js('Katie'), Js('Kayla'), Js('Kayleigh'), Js('Keira'), Js('Lacey'), Js('Lara'), Js('Laura'), Js('Lauren'), Js('Layla'), Js('Leah'), Js('Lexi'), Js('Lexie'), Js('Libby'), Js('Lilly'), Js('Lily'), Js('Lola'), Js('Louise'), Js('Lucy'), Js('Lydia'), Js('Maddison'), Js('Madeleine'), Js('Madison'), Js('Maisie'), Js('Maisy'), Js('Maria'), Js('Martha'), Js('Matilda'), Js('Maya'), Js('Megan'), Js('Melissa'), Js('Mia'), Js('Millie'), Js('Mollie'), Js('Molly'), Js('Morgan'), Js('Mya'), Js('Naomi'), Js('Natasha'), Js('Niamh'), Js('Nicole'), Js('Olivia'), Js('Paige'), Js('Phoebe'), Js('Poppy'), Js('Rachel'), Js('Rebecca'), Js('Rose'), Js('Rosie'), Js('Ruby'), Js('Samantha'), Js('Sara'), Js('Sarah'), Js('Scarlett'), Js('Shannon'), Js('Sienna'), Js('Skye'), Js('Sofia'), Js('Sophia'), Js('Sophie'), Js('Summer'), Js('Tegan'), Js('Tia'), Js('Tilly'), Js('Victoria'), Js('Willow'), Js('Yasmin'), Js('Zara'), Js('Zoe')]))
var.put('nm3', Js([Js('Aaren'), Js('Addison'), Js('Aiden'), Js('Alex'), Js('Alexis'), Js('Ali'), Js('Angel'), Js('Ash'), Js('Ashley'), Js('Ashton'), Js('Aubrey'), Js('Avery'), Js('Bailey'), Js('Bennie'), Js('Bev'), Js('Billie'), Js('Billy'), Js('Blair'), Js('Blake'), Js('Bret'), Js('Brett'), Js('Brice'), Js('Brook'), Js('Brynn'), Js('Caden'), Js('Cameron'), Js('Carmen'), Js('Carol'), Js('Casey'), Js('Charlie'), Js('Chris'), Js('Clem'), Js('Cory'), Js('Dane'), Js('Danni'), Js('Danny'), Js('Denny'), Js('Drew'), Js('Eli'), Js('Elliot'), Js('Emerson'), Js('Erin'), Js('Fran'), Js('Frankie'), Js('Franky'), Js('Gabby'), Js('Gabe'), Js('Gail'), Js('Gale'), Js('Gene'), Js('Glen'), Js('Glenn'), Js('Haiden'), Js('Harley'), Js('Harper'), Js('Hayden'), Js('Jackie'), Js('Jaden'), Js('Jaime'), Js('Jamie'), Js('Jess'), Js('Jesse'), Js('Jessie'), Js('Jo'), Js('Jody'), Js('Jordan'), Js('Jude'), Js('Justice'), Js('Kai'), Js('Kerry'), Js('Kiran'), Js('Kit'), Js('Kris'), Js('Lane'), Js('Lee'), Js('Leigh'), Js('Lesley'), Js('Leslie'), Js('Logan'), Js('Lynn'), Js('Maddox'), Js('Marley'), Js('Mason'), Js('Mel'), Js('Mell'), Js('Morgan'), Js('Nicky'), Js('Noel'), Js('Phoenix'), Js('Quinn'), Js('Ray'), Js('Raylee'), Js('Reed'), Js('Reggie'), Js('Rene'), Js('Riley'), Js('River'), Js('Robin'), Js('Rory'), Js('Rowan'), Js('Rudy'), Js('Ryan'), Js('Sam'), Js('Sammy'), Js('Shay'), Js('Sidney'), Js('Silver'), Js('Skye'), Js('Skylar'), Js('Skyler'), Js('Steff'), Js('Tanner'), Js('Taylor'), Js('Terry'), Js('Tyler'), Js('Val'), Js('Vic'), Js('Will'), Js('Willy')]))
var.put('nm4', Js([Js('Alder'), Js('Amber'), Js('Ash'), Js('Autumn'), Js('Beau'), Js('Belle'), Js('Black'), Js('Bran'), Js('Bright'), Js('Cart'), Js('Clear'), Js('Cliff'), Js('Cloud'), Js('Cold'), Js('Common'), Js('Coven'), Js('Crag'), Js('Dawn'), Js('Deep'), Js('Dew'), Js('Distant'), Js('Dream'), Js('Dusk'), Js('Earth'), Js('Elder'), Js('Ember'), Js('Even'), Js('Fair'), Js('Far'), Js('Feather'), Js('Fern'), Js('Fire'), Js('Flame'), Js('Flat'), Js('Forest'), Js('Free'), Js('Full'), Js('Glad'), Js('Glory'), Js('Gold'), Js('Grand'), Js('Gray'), Js('Great'), Js('Hard'), Js('Haven'), Js('Hay'), Js('Heart'), Js('Heavy'), Js('High'), Js('Hill'), Js('Honor'), Js('Humble'), Js('Keen'), Js('Kings'), Js('Light'), Js('Lone'), Js('Long'), Js('Low'), Js('Meadow'), Js('Merry'), Js('Mid'), Js('Mild'), Js('Moon'), Js('Morning'), Js('Night'), Js('Noble'), Js('Pen'), Js('Plain'), Js('Pont'), Js('Pride'), Js('Proud'), Js('Rain'), Js('Raven'), Js('Regal'), Js('River'), Js('Rose'), Js('Sage'), Js('Shadow'), Js('Sharp'), Js('Shield'), Js('Silent'), Js('Silver'), Js('Simple'), Js('Single'), Js('Sky'), Js('Snow'), Js('Soft'), Js('Spirit'), Js('Spring'), Js('Star'), Js('Stark'), Js('Stern'), Js('Still'), Js('Stout'), Js('Strong'), Js('Summer'), Js('Sun'), Js('Swift'), Js('Tall'), Js('Town'), Js('Tree'), Js('True'), Js('Way'), Js('Whisper'), Js('Whit'), Js('White'), Js('Wild'), Js('Wind'), Js('Winter'), Js('Wise'), Js('Wolf'), Js('Wood'), Js('Young')]))
var.put('nm5', Js([Js('bane'), Js('beam'), Js('bloom'), Js('blossom'), Js('bluff'), Js('born'), Js('bough'), Js('bow'), Js('braid'), Js('branch'), Js('brand'), Js('breath'), Js('breeze'), Js('brook'), Js('brow'), Js('child'), Js('creek'), Js('crest'), Js('cross'), Js('dale'), Js('dew'), Js('down'), Js('draft'), Js('dream'), Js('fall'), Js('flare'), Js('flaw'), Js('fleur'), Js('flow'), Js('flower'), Js('force'), Js('ford'), Js('gaze'), Js('gem'), Js('glade'), Js('gleam'), Js('glide'), Js('grove'), Js('guard'), Js('hallow'), Js('heart'), Js('hood'), Js('keep'), Js('lace'), Js('land'), Js('law'), Js('less'), Js('man'), Js('mark'), Js('mercy'), Js('might'), Js('mill'), Js('more'), Js('peace'), Js('ridge'), Js('root'), Js('run'), Js('scar'), Js('send'), Js('shade'), Js('sky'), Js('smith'), Js('soar'), Js('song'), Js('spark'), Js('stern'), Js('stone'), Js('stride'), Js('sun'), Js('surge'), Js('sworn'), Js('thorn'), Js('tide'), Js('tower'), Js('tree'), Js('vale'), Js('ward'), Js('water'), Js('weather'), Js('weaver'), Js('well'), Js('whirl'), Js('wind'), Js('winds'), Js('wine'), Js('wing'), Js('winter'), Js('wood'), Js('woods'), Js('wright')]))
pass
pass


# Add lib to the module scope
mortalInstruments = var.to_python()

File: js/mortalInstruments.js
var nm1 = ["Aaron","Adam","Aidan","Aiden","Alex","Alexander","Alfie","Andrew","Anthony","Archie","Arthur","Ashton","Bailey","Ben","Benjamin","Billy","Blake","Bobby","Bradley","Brandon","Caleb","Callum","Cameron","Charles","Charlie","Christopher","Cody","Connor","Corey","Daniel","David","Declan","Dexter","Dominic","Dylan","Edward","Elliot","Ellis","Ethan","Evan","Ewan","Finlay","Finley","Frankie","Freddie","Frederick","Gabriel","George","Harley","Harrison","Harry","Harvey","Hayden","Henry","Isaac","Jack","Jackson","Jacob","Jake","James","Jamie","Jay","Jayden","Jenson","Joe","Joel","John","Jonathan","Jordan","Joseph","Josh","Joshua","Jude","Kai","Kayden","Kian","Kieran","Kyle","Leo","Leon","Lewis","Liam","Logan","Louie","Louis","Luca","Lucas","Luke","Mason","Matthew","Max","Michael","Morgan","Nathan","Nicholas","Noah","Oliver","Ollie","Oscar","Owen","Patrick","Peter","Reece","Reuben","Rhys","Riley","Robert","Rory","Ryan","Sam","Samuel","Scott","Sean","Sebastian","Spencer","Stanley","Taylor","Theo","Thomas","Toby","Tom","Tommy","Tyler","William","Zac","Zachary","Zak"];
var nm2 = ["Abbie","Abby","Abigail","Aimee","Alexandra","Alice","Alicia","Alisha","Amber","Amelia","Amelie","Amy","Anna","Ava","Bella","Bethany","Brooke","Caitlin","Cerys","Charlie","Charlotte","Chelsea","Chloe","Courtney","Daisy","Danielle","Demi","Eleanor","Eliza","Elizabeth","Ella","Ellie","Eloise","Elsie","Emilia","Emily","Emma","Erin","Esme","Eva","Eve","Evelyn","Evie","Faith","Freya","Georgia","Georgina","Grace","Gracie","Hannah","Harriet","Heidi","Hollie","Holly","Imogen","Isabel","Isabella","Isabelle","Isla","Isobel","Jade","Jasmine","Jennifer","Jessica","Jodie","Julia","Kate","Katherine","Katie","Kayla","Kayleigh","Keira","Lacey","Lara","Laura","Lauren","Layla","Leah","Lexi","Lexie","Libby","Lilly","Lily","Lola","Louise","Lucy","Lydia","Maddison","Madeleine","Madison","Maisie","Maisy","Maria","Martha","Matilda","Maya","Megan","Melissa","Mia","Millie","Mollie","Molly","Morgan","Mya","Naomi","Natasha","Niamh","Nicole","Olivia","Paige","Phoebe","Poppy","Rachel","Rebecca","Rose","Rosie","Ruby","Samantha","Sara","Sarah","Scarlett","Shannon","Sienna","Skye","Sofia","Sophia","Sophie","Summer","Tegan","Tia","Tilly","Victoria","Willow","Yasmin","Zara","Zoe"];
var nm3 = ["Aaren","Addison","Aiden","Alex","Alexis","Ali","Angel","Ash","Ashley","Ashton","Aubrey","Avery","Bailey","Bennie","Bev","Billie","Billy","Blair","Blake","Bret","Brett","Brice","Brook","Brynn","Caden","Cameron","Carmen","Carol","Casey","Charlie","Chris","Clem","Cory","Dane","Danni","Danny","Denny","Drew","Eli","Elliot","Emerson","Erin","Fran","Frankie","Franky","Gabby","Gabe","Gail","Gale","Gene","Glen","Glenn","Haiden","Harley","Harper","Hayden","Jackie","Jaden","Jaime","Jamie","Jess","Jesse","Jessie","Jo","Jody","Jordan","Jude","Justice","Kai","Kerry","Kiran","Kit","Kris","Lane","Lee","Leigh","Lesley","Leslie","Logan","Lynn","Maddox","Marley","Mason","Mel","Mell","Morgan","Nicky","Noel","Phoenix","Quinn","Ray","Raylee","Reed","Reggie","Rene","Riley","River","Robin","Rory","Rowan","Rudy","Ryan","Sam","Sammy","Shay","Sidney","Silver","Skye","Skylar","Skyler","Steff","Tanner","Taylor","Terry","Tyler","Val","Vic","Will","Willy"];
var nm4 = ["Alder","Amber","Ash","Autumn","Beau","Belle","Black","Bran","Bright","Cart","Clear","Cliff","Cloud","Cold","Common","Coven","Crag","Dawn","Deep","Dew","Distant","Dream","Dusk","Earth","Elder","Ember","Even","Fair","Far","Feather","Fern","Fire","Flame","Flat","Forest","Free","Full","Glad","Glory","Gold","Grand","Gray","Great","Hard","Haven","Hay","Heart","Heavy","High","Hill","Honor","Humble","Keen","Kings","Light","Lone","Long","Low","Meadow","Merry","Mid","Mild","Moon","Morning","Night","Noble","Pen","Plain","Pont","Pride","Proud","Rain","Raven","Regal","River","Rose","Sage","Shadow","Sharp","Shield","Silent","Silver","Simple","Single","Sky","Snow","Soft","Spirit","Spring","Star","Stark","Stern","Still","Stout","Strong","Summer","Sun","Swift","Tall","Town","Tree","True","Way","Whisper","Whit","White","Wild","Wind","Winter","Wise","Wolf","Wood","Young"];
var nm5 = ["bane","beam","bloom","blossom","bluff","born","bough","bow","braid","branch","brand","breath","breeze","brook","brow","child","creek","crest","cross","dale","dew","down","draft","dream","fall","flare","flaw","fleur","flow","flower","force","ford","gaze","gem","glade","gleam","glide","grove","guard","hallow","heart","hood","keep","lace","land","law","less","man","mark","mercy","might","mill","more","peace","ridge","root","run","scar","send","shade","sky","smith","soar","song","spark","stern","stone","stride","sun","surge","sworn","thorn","tide","tower","tree","vale","ward","water","weather","weaver","well","whirl","wind","winds","wine","wing","winter","wood","woods","wright"];
var br = "";

function nameGen(type){
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd2 = Math.floor(Math.random() * nm4.length);
		rnd3 = Math.floor(Math.random() * nm5.length);
		nl = nm4[rnd2] + nm5[rnd3];
		if(tp === 1){
			rnd = Math.floor(Math.random() * nm2.length);
			names = nm2[rnd] + " " + nl;
		}else if(tp === 2){
			rnd = Math.floor(Math.random() * nm3.length);
			names = nm3[rnd] + " " + nl;
		}else{
			rnd = Math.floor(Math.random() * nm1.length);
			names = nm1[rnd] + " " + nl;
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}